src/algebra/squarefree.lean

```diff
@@ -85,6 +85,42 @@ end
 
 end multiplicity
 
+lemma eq_zero_or_no_irreducibles_and_squarefree_iff_irreducible_sq_not_dvd
```

Comment:
This lemma maybe makes more sense to me phrased in the other direction, as a characterization of when a square of an irreducible divides an element, because it doesn't seem like a great characterization of squarefreeness with all the other data.

Reply:
Also, for several of these lemmas - is there anything we can say about a `wf_dvd_monoid` without irreducibles? If it doesn't hit these requirements, it's a group with zero, and I guess there's no ''isn't a group with zero'' class...

Reply:
Sure, I'll swap the order! I'm not sure for your second question to be honest - this area of maths isn't my strong suit, I just needed these lemmas...

Reply:
I've swapped the order - what do you think the best option is for your remaining concern?

Reply:
@awainverse :ping_pong: